pkg/ips/primarynetwork_test.go

```diff
@@ -0,0 +1,231 @@
+package ips
+
+import (
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/kubevirt/ipam-extensions/pkg/config"
+)
+
+var _ = Describe("VmiInterfaceIPRequests", func() {
+	var (
+		vmi        *virtv1.VirtualMachineInstance
+		primaryNet *config.RelevantConfig
+	)
+
+	BeforeEach(func() {
+		vmi = &virtv1.VirtualMachineInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-vm",
+				Namespace: "default",
+			},
+		}
+
+		primaryNet = &config.RelevantConfig{
+			Name:               "primarynet",
+			AllowPersistentIPs: true,
+			Role:               config.NetworkRolePrimary,
+			Subnets:            "192.168.0.0/16,fd12:1234::/64",
+		}
+	})
+
+	When("VMI has no IP requests annotation", func() {
+		It("should return nil when no IP requests are present", func() {
+			result, err := VmiInterfaceIPRequests(vmi, "podnet", primaryNet)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(BeNil())
+		})
+	})
+
+	When("VMI has IP requests annotation", func() {
+		Context("with valid IPv4 and IPv6 addresses", func() {
+			BeforeEach(func() {
+				ipRequests := map[string][]string{
+					"podnet": {"192.168.1.10", "fd20:1234::200"},
```

Comment:
notice that "fd20:1234::200" is not in "fd12:1234::/64" subnet that is set [here](https://github.com/kubevirt/ipam-extensions/pull/105/files#diff-d3e654871f4829cf4633471bb57cb0c485f0839164a9231e9f24bb79ebd4d721R33). 
The configuration seems invalid.
Question is - should we just be a vessel that copies the kubevirt annotation to multus annotation and let OVNK reject this?

Reply:
In terms of visibility I agree the error must be seen by the user where they expect it (i.e. the pod and ipamClaim).
You can resolve this.